core/drivers/pcap.h

```diff
@@ -3,17 +3,36 @@
 #include <pcap/pcap.h>
 
 #include <glog/logging.h>
-
+#include "pcap.h"
 #include "../message.h"
 #include "../port.h"
 #include "../snobj.h"
 #include "../utils/pcap.h"
 
 /*!
- * Queue of data to be transmitted. 
- * TODO: Why is this global and not declared on the stack?
+ * All problems in computer science can be solved by another layer of
+ * indirection. This class wraps all function calls to libpcap from PCAPPort.
+ * Serves two purposes -- (1) if libpcap functions change, we can rename them
+ * here and not throughout the PCAPPort driver (2) in test mode, can swap this
+ * class out with a mock to interpose on parameters PCAPPort is passing to/from
+ * libpcap.
  */
-static unsigned char tx_pcap_data[PCAP_SNAPLEN];
+class LibPCAP {
```

Comment:
Use of mock objects must be done with care, since virtual functions degrade performance. While it is fine for 'slow' device drivers, such as PCAP, this approach should not apply to 'fast' devices, such as PMD and VPort.